app/main.py

```diff
@@ -1,13 +1,179 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
+
+
+error = {
+    "code": "E501",
+    "filename": "./source_code_2.py",
+    "line_number": 18,
+    "column_number": 80,
+    "text": "line too long (99 > 79 characters)",
+    "physical_line": """    return f"I like to filter, rounding, doubling,"""
+                     """store and decorate numbers: {", ".join(items)}!\"""",
+}
+
+print(format_linter_error(error=error))
+
+
+# The output will be:
+# {
+#     "line": 18,
+#     "column": 80,
+#     "message": "line too long (99 > 79 characters)",
+#     "name": "E501",
+#     "source": "flake8"
+# }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors":
+            [
+                {
+                    "line": error["line_number"],
+                    "column": error["column_number"],
+                    "message": error["text"],
+                    "name": error["code"],
+                    "source": "flake8"
+                } for error in errors
+            ],
+        "path": file_path,
+        "status": "failed"
```

Comment:
You should return failed if `errors` is not empty. You should return `passed` if `errors` is empty
```suggestion
        "status": "failed" if errors else "passed"
```